includes/multi-currency/class-frontend-prices.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * WooCommerce Payments Multi Currency Frontend Prices
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Multi_Currency;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that applies Multi Currency prices on the frontend.
+ */
+class Frontend_Prices {
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var Multi_Currency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 */
+	public function __construct( Multi_Currency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+
+		// Simple product price hooks.
+		add_filter( 'woocommerce_product_get_price', [ $this, 'get_product_price' ] );
+		add_filter( 'woocommerce_product_get_regular_price', [ $this, 'get_product_price' ] );
+		add_filter( 'woocommerce_product_get_sale_price', [ $this, 'get_product_price' ] );
+
+		// Variation price hooks.
+		add_filter( 'woocommerce_product_variation_get_price', [ $this, 'get_product_price' ] );
+		add_filter( 'woocommerce_product_variation_get_regular_price', [ $this, 'get_product_price' ] );
+		add_filter( 'woocommerce_product_variation_get_sale_price', [ $this, 'get_product_price' ] );
+
+		// Variation price range hooks.
+		add_filter( 'woocommerce_variation_prices', [ $this, 'get_variation_price_range' ] );
+		add_filter( 'woocommerce_get_variation_prices_hash', [ $this, 'add_exchange_rate_to_variation_prices_hash' ] );
+
+		// Shipping methods hooks.
+		add_filter( 'woocommerce_package_rates', [ $this, 'get_shipping_rates_prices' ] );
+
+		// Coupon hooks.
+		add_filter( 'woocommerce_coupon_get_amount', [ $this, 'get_coupon_amount' ], 20, 2 );
+		add_filter( 'woocommerce_coupon_get_minimum_amount', [ $this, 'get_coupon_min_max_amount' ] );
+		add_filter( 'woocommerce_coupon_get_maximum_amount', [ $this, 'get_coupon_min_max_amount' ] );
```

Comment:
Should we register these hooks with a higher priority to ensure they're executed after the default priority?

Reply:
As with the other hooks, I think this sounds sensible.

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/1860/commits/950f6095631fdd12d953cc7d09b7353a042ccb1d